packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1943,47 +1943,43 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin {
            _textLayoutLastMinWidth == constraints.minWidth,
       'Last width ($_textLayoutLastMinWidth, $_textLayoutLastMaxWidth) not the same as max width constraint (${constraints.minWidth}, ${constraints.maxWidth}).');
     final TextRange word = _textPainter.getWordBoundary(position);
+    final String originalText = textSelectionDelegate.textEditingValue.text;
     // When long-pressing past the end of the text, we want a collapsed cursor.
     if (position.offset >= word.end)
       return TextSelection.fromPosition(position);
     // If text is obscured, the entire sentence should be treated as one word.
     if (obscureText) {
       return TextSelection(baseOffset: 0, extentOffset: _plainText.length);
     // If the word is a space, on iOS try to select the previous word instead.
-    } else if (text?.text != null
-        && _isWhitespace(text!.text!.codeUnitAt(position.offset))
-        && position.offset > 0) {
+    // On Android try to select the previous word instead only if the text is read only.
+    } else if (position.offset > 0 && ((text?.text != null
```

Comment:
if you do the effectiveText, you can keep the same if case but replace text?.text with effectiveText

Reply:
edit: the original else if should just be text?.toPlainText() instead